actors/builtin/miner/miner_state.go

```diff
@@ -72,11 +71,12 @@ type WorkerKeyChange struct {
 }
 
 type SectorPreCommitInfo struct {
-	SectorNumber abi.SectorNumber
-	SealedCID    cid.Cid // CommR
-	SealEpoch    abi.ChainEpoch
-	DealIDs      []abi.DealID
-	Expiration   abi.ChainEpoch // Sector Expiration
+	RegisteredProof abi.RegisteredProof
```

Comment:
allowed to add here @porcuquine ? I think we want it included with precommit so we can know when the PreCommit should expire if the SNARK doesn't appear on chain...?

Reply:
Is it not sufficient to put it in the cron callback payload? Even there I think it's redundant because it's implied by the epoch that the event is scheduled for.

Putting it here is expensive - it must live for a long time on chain, way past its period of utility.

Reply:
point well taken. Fixed

Reply:
Thank you. I should have noted is that an alternative is to restructure the on-chain structures so that the committed sector doesn't include the whole precommitted sector into, but extracts only the fields necessary. Arguably we should do that anyway. cc @magik6k

Reply:
will leave for future PR. If you want to link me to area this is done in Lotus @magik6k I can PR

Reply:
Per convo with @whyrusleeping I think it does have to be on chain since it is part of a per sector configuration... Not sure what truth is here.

Reply:
- it is not sufficient to store this in the callback, as the sector must be tied to the instance
- this should be very cheap as it is a single number
- all other places MUST look this up here and refer to it when interacting with this sector